### DISCUSSION TITLE
Replays can have a wrong offset applied to them causing major inconsistencies

### DISCUSSION THREAD

**Discussion by WitherFlower** [ppy/osu#17362] [Q&A] [answered]
_I mentioned in the meeting that there might be some gameplay differences in how stacked sliders are handled with classic mod, but I found out another issue while looking for the slider one. Will continue to investigate on that._

I observed this phenomenon on a few maps, mostly old maps so the issue might be linked to some kind of offset specific to older formats not being applied in lazer.

1st map : https://osu.ppy.sh/beatmapsets/243#osu/1147
| stable | lazer after watching replay |
| :---: | :---: |
| ![screenshot502](https://user-images.githubusercontent.com/52672543/159153401-f27ff5d0-2713-46ff-aa50-875084f3f91e.jpg) | ![image](https://user-images.githubusercontent.com/52672543/159153314-988caaca-b49f-4fc0-86c1-b763a00c17e8.png) |

Replay : https://drive.google.com/file/d/1SeHl50mi_wQon-qKX_WbzOStRN6dC5dE/view?usp=sharing

2nd map : https://osu.ppy.sh/beatmapsets/131#osu/292
| stable | lazer after watching replay |
| :---: | :---: |
| ![screenshot503](https://user-images.githubusercontent.com/52672543/159153764-c6a07304-cdb3-492b-b4dd-ada76881d6b6.jpg) | ![image](https://user-images.githubusercontent.com/52672543/159153700-543fc0a2-45be-4fba-9190-fd4077bb2d01.png) |

Replay : https://drive.google.com/file/d/1zNIcabn259aHzJTnCu1oZYFUOU4VBr-_/view?usp=sharing

**Comment by peppy** [comment, marked as answer]
The issue here looks to be to do with the 24ms offset from `BEATMAP_VERSION` 5 and higher (ie. there's an offset applied to earlier versions). It seems to be applied to hitobjects themselves in the decoder, but in stable it looks to be applied only in the editor.. needs further investigation into the stable code to see which is more correct. Or more importantly, how these replays are working on stable in the first place.

It seems like we may *not* want to apply the offset the way we are?

**Reply by peppy** [reply]
Tracking at https://github.com/ppy/osu/issues/17404. This is a pretty bad issue and going to require some further discussion to come to a resolution, but TLDR: stable replays are correct and will be handled correctly after we fix the issue (but any replays created on these beatmaps in lazer are incorrect and will be broken once we fix the underlying issue).

**Comment by bdach** [comment]
#17429 has improved things some, but not completely.

The first map replay (https://osu.ppy.sh/beatmapsets/243#osu/1147) still looks to have a strange slider miss in it:

![osu_2022-03-24_23-13-42](https://user-images.githubusercontent.com/20418176/160019966-9aa7a439-ee6e-4fb8-b286-01e955ffed6c.jpg)

and the second map replay (https://osu.ppy.sh/beatmapsets/131#osu/292) exchanges one 100 for one 300 after the replay is played back, but that'll 99% be down to #14573.

**Reply by WitherFlower** [reply]
I already had a feeling that this kickslider stack played differently on lazer+classic than on stable, and this is why I mentioned the "difference in gameplay for slider stacks" in last meeting (This is also the reason why I looked at replays of this map, but ended up finding the other bug). 
I'll investigate further, but this seems like an example of that differing behavior.
